examples/notification-demo/echo_server/echo.c

```diff
@@ -0,0 +1,1975 @@
+    circlog_size = (size_t)circlog_size_uint64;
+    if (ret < 0) {
+        fprintf(stderr, "%s L%d: wolfsentry_user_value_get_string(\"circlog-size\") returned "
+                WOLFSENTRY_ERROR_FMT "\n",
+                __FILE__, __LINE__,
+                WOLFSENTRY_ERROR_FMT_ARGS(ret));
+        exit(1);
+    }
+
+    ret = circlog_init(circlog_size);
+    if (ret < 0) {
+        fprintf(stderr, "circlog_init() failed: " WOLFSENTRY_ERROR_FMT "\n", WOLFSENTRY_ERROR_FMT_ARGS(ret));
+        exit(1);
+    }
+
+#ifdef CIRCLOG_UNIT_TEST
+{
+    char *msg_buf;
+    struct circlog_message *msg;
+    int i;
+
+    static const char text_s[] = "I'm a little teapot short and stout.";
+
+    ret = circlog_enqueue_one(strlen(text_s), &msg_buf);
+    if (ret < 0) {
+        fprintf(stderr, " %s L%d circlog failed: " WOLFSENTRY_ERROR_FMT "\n", __FILE__, __LINE__, WOLFSENTRY_ERROR_FMT_ARGS(ret));
+        exit(1);
+    }
+
+    memcpy(msg_buf, text_s, strlen(text_s));
+
+    ret = circlog_dequeue_one(&msg);
+    if (ret < 0) {
+        fprintf(stderr, " %s L%d circlog failed: " WOLFSENTRY_ERROR_FMT "\n", __FILE__, __LINE__, WOLFSENTRY_ERROR_FMT_ARGS(ret));
+        exit(1);
+    }
+
+    printf("%.*s\n",(int)msg->len,msg->msg_buf);
+
+    fprintf(stderr,"circlog_head=%zu circlog_tail=%zu\n",circlog_head,circlog_tail);
+
+    for (i = 0; i < 3 * (circlog_size / circlog_ent_size_from_msg_len(strlen(text_s))); ++i) {
+        size_t circlog_head_after_enqueue,circlog_tail_after_enqueue;
+
+        ret = circlog_enqueue_one(strlen(text_s), &msg_buf);
+        if (ret < 0) {
+            fprintf(stderr, " %s L%d circlog failed: " WOLFSENTRY_ERROR_FMT "\n", __FILE__, __LINE__, WOLFSENTRY_ERROR_FMT_ARGS(ret));
+            exit(1);
+        }
+
+        circlog_head_after_enqueue = circlog_head;
+        circlog_tail_after_enqueue = circlog_tail;
+
+        memcpy(msg_buf, text_s, strlen(text_s));
+
+        ret = circlog_dequeue_one(&msg);
+        if (ret < 0) {
+            fprintf(stderr, " %s L%d circlog failed: " WOLFSENTRY_ERROR_FMT "\n", __FILE__, __LINE__, WOLFSENTRY_ERROR_FMT_ARGS(ret));
+            exit(1);
+        }
+
+        if (((size_t)msg->len != strlen(text_s)) || (memcmp(msg->msg_buf, text_s, msg->len) != 0)) {
+            fprintf(stderr, "circlog msg doesn't match what went in, i=%d msg->len=%u circlog_head_after_enqueue=%zu circlog_tail_after_enqueue=%zu circlog_head=%zu circlog_tail=%zu.\n",i,msg->len,circlog_head_after_enqueue,circlog_tail_after_enqueue,circlog_head,circlog_tail);
+            exit(1);
+        }
+    }
+
+
+    fprintf(stderr,"circlog_head=%zu circlog_tail=%zu\n",circlog_head,circlog_tail);
+}
+#endif /* CIRCLOG_UNIT_TEST */
+
+
+  echo_init();
+
+  /* Infinite loop */
+  for (;;) {
+    int peer_fd;
+    struct sockaddr_in server_addr;
+    socklen_t server_len;
+    struct sockaddr_in client_addr;
+    socklen_t client_len;
+    struct wolfsentry_data *wolfsentry_data = NULL;
+    int ret;
+#ifdef HTTP_NONBLOCKING
```

Comment:
note I haven't tested `HTTP_NONBLOCKING` in this -- it's residual from Andrew's work with LWIP.